### PR TITLE
Tree documentation

### PR DESCRIPTION
**Pull request by adityagoel4512**
Documented trees

### CODE REVIEW

**Review by rbanerjee20 — CHANGES_REQUESTED** [review]
Looks good overall! Could we add one tree example at the end, just to give an overview of how these methods and constructors can be used together? 

Might be a good plan to add them to the other data structures too as a separate ticket.

File: source/tutorial.rst [hunk 1]
```diff
@@ -461,6 +461,62 @@ Swaps the elements at ``index1`` and ``index2`` in the array. The optional ``lon
 
 Returns the fixed size of the array.
 
+Tree<T>
+###############
+
+The tree type encapsulates an underlying Node<T> object. The distinction between the Node and Tree types exist to allow you to be able to specify exactly which nodes in a subtree should be animated.
```
Comment (iz0825) [inline source/tutorial.rst]: Really minor grammar thing - don't think we need both "allow" and "be able to" here - just to make the sentence more straight forward.

**Comment by adityagoel4512** [comment]
> Looks good overall! Could we add one tree example at the end, just to give an overview of how these methods and constructors can be used together?
> 
> Might be a good plan to add them to the other data structures too as a separate ticket.

What sort of thing would be in the example?

**Comment by rbanerjee20** [comment]
>
> > Looks good overall! Could we add one tree example at the end, just to give an overview of how these methods and constructors can be used together?
> > Might be a good plan to add them to the other data structures too as a separate ticket.
> 
> What sort of thing would be in the example?

Just a few lines showing the typical usage? E.g.,

```
let head = Node<number>(4);
let tree = Tree<Node<number>>(head);
head.left = buildBST(0, 2, arr);
head.right = buildBST(4, arr.size()-1, arr);
```

which is taken from one of our examples (but without the recursive calls).

**Review by rbanerjee20 — APPROVED** [review]

**Pull request merged**